src/generic/Matrix.jl

```diff
@@ -201,12 +197,37 @@ Base.@propagate_inbounds function setindex!(a::MatrixElem, d::T, r::Int,
     a.entries[r, c] = base_ring(a)(d)
 end
 
+Base.isassigned(a::Union{Mat,MatAlgElem}, i, j) = isassigned(a.entries, i, j)
+
+function Base.isassigned(a::MatrixElem, i, j)
+    try
+        a[i, j]
+        true
+    catch e
+        if isa(e, BoundsError) || isa(e, UndefRefError)
+            return false
+        else
+            rethrow()
+        end
+    end
+end
+
 @doc Markdown.doc"""
     zero(a::AbstractAlgebra.MatSpace)
 > Construct the zero matrix in the given matrix space.
 """
 zero(a::AbstractAlgebra.MatSpace) = a()
 
+zero(x::MatrixElem) = zero(parent(x))
```

Comment:
I know that this is not clearly written somewhere, but we introduced the `similar` functions to get rid of all the `parent` calls in the matrix code. (I don't want to go into details of why we want to minimize the calls to `parent`).

It would be good if we can we make the `zero(::MatrixElem)` function not use `parent`.